DEVELOPER_GUIDE.md

```diff
@@ -104,6 +103,30 @@ To check the markdown files, run the following command:
 make lint.markdown
 ```
 
+### Golang lint
```

Comment:
Add these to TOC? Capitalize Like The Rest of It?

Reply:
Add my changes to TOC. Also added a vim section to Editors section and changed the golangci-linter in the Makefile to run via Docker, so users don't need to install it first and we all run the same linter version.